Gemfile

```diff
@@ -84,12 +73,31 @@ if RUBY_PLATFORM != 'java'
   end
 end
 
-# For type checking
-# Sorbet releases almost daily, with new checks introduced that can make a
-# previously-passing codebase start failing. Thus, we need to lock to a specific
-# version and bump it from time to time.
-# Also, there's no support for windows
-if RUBY_VERSION >= '2.4.0' && (RUBY_PLATFORM =~ /^x86_64-(darwin|linux)/)
-  gem 'sorbet', '= 0.5.9672'
-  gem 'spoom', '~> 1.1'
+group :appraisal do
+  gem 'appraisal', '~> 2.2'
+end
+
+group :release do
+  gem 'pimpmychangelog', '>= 0.1.2' # Formats the CHANGELOG.md file
+  gem 'redcarpet', '~> 3.4' if RUBY_PLATFORM != 'java' # Used by YARD to generate docs
+  gem 'yard', '~> 0.9'
+end
```

Comment:
### Approach 2

Locally grouped gems, matching the high-level activity they are part of: e.g. `release`, `lint`, `benchmark`, `test`, etc.

Reply:
### Approach 2a

Locally grouped gems, but more granular then `2`: e.g. `changelog` (has `pimpmychangelog`), `docs` (has `redcarpet` & `yard`), `lint` (has `rubocop*`), `type_check` (has `sorbet` & `spoom`) etc.

Reply:
I don't have strong thoughts either way, but I think coarser-grained groups seem like a good enough start, and we could always break them up as needed later.